config/config.go

```diff
@@ -850,92 +857,50 @@ func (c *Config) addInput(name string, table *ast.Table) error {
 	// If the input has a SetParser or SetParserFunc function, it can accept
 	// arbitrary data-formats, so build the requested parser and set it.
 	if t, ok := input.(telegraf.ParserInput); ok {
-		missThreshold = 1
-		if parser, err := c.addParser(name, table); err == nil {
-			t.SetParser(parser)
-		} else {
-			missThreshold = 0
-			// Fallback to the old way of instantiating the parsers.
-			config, err := c.getParserConfig(name, table)
-			if err != nil {
-				return err
-			}
-			parser, err := c.buildParserOld(name, config)
-			if err != nil {
-				return err
-			}
-			t.SetParser(parser)
+		parser, err := c.addParser(name, table)
+		if err != nil {
+			return fmt.Errorf("adding parser failed: %w", err)
 		}
+		t.SetParser(parser)
 	}
 
 	// Keep the old interface for backward compatibility
 	if t, ok := input.(parsers.ParserInput); ok {
 		// DEPRECATED: Please switch your plugin to telegraf.ParserInput.
-		missThreshold = 1
-		if parser, err := c.addParser(name, table); err == nil {
-			t.SetParser(parser)
-		} else {
-			missThreshold = 0
-			// Fallback to the old way of instantiating the parsers.
-			config, err := c.getParserConfig(name, table)
-			if err != nil {
-				return err
-			}
-			parser, err := c.buildParserOld(name, config)
-			if err != nil {
-				return err
-			}
-			t.SetParser(parser)
+		parser, err := c.addParser(name, table)
+		if err != nil {
+			return fmt.Errorf("adding parser failed: %w", err)
 		}
+		t.SetParser(parser)
 	}
```

Comment:
I think everything is switched over to telegraf.ParserInput and this is the only reference to **parsers**.ParserInput now. Should we remove this and also remove it in plugins/parsers/registry.go?

Reply:
I thought removing `parsers.ParserInput` might break external plugins and thus kept it. Happy to remove it if you think that's not a concern...

Reply:
That makes sense. It's not a problem to keep this until a major release when we can change the interface.